ocaml/xapi/xapi.ml

```diff
@@ -1129,23 +1085,132 @@ let server_init () =
             Helpers.touch_file !Xapi_globs.ready_file ;
             (* Keep trying to log into master *)
             let finished = ref false in
+            let xapi_ver_high_alerted = ref false in
+
             while not !finished do
               (* Grab the management IP address (wait forever for it if necessary) *)
               let ip = wait_for_management_ip_address ~__context in
               debug "Start master_connection watchdog" ;
               ignore (Master_connection.start_master_connection_watchdog ()) ;
               debug "Attempting to communicate with master" ;
-              (* Try to say hello to the pool *)
-              match attempt_pool_hello ip with
+
+              (* Attempt checking host status with pool coordinator:
```

Comment:
The code lines enclosed by the `while` loop look too many.
Would suggest to keep the original structure, with a function being separated.

Reply:
OK, moved out.